sycl/source/detail/kernel_impl.cpp

```diff
@@ -106,6 +106,38 @@ void kernel_impl::checkIfValidForNumArgsInfoQuery() const {
       "interoperability function or to query a device built-in kernel");
 }
 
+bool kernel_impl::exceedsOccupancyResourceLimits(
+    const device &Device, const range<3> &WorkGroupSize,
+    size_t DynamicLocalMemorySize) const {
+  // Respect occupancy limits for WorkGroupSize and DynamicLocalMemorySize.
+  // Generally, exceeding hardware resource limits will yield in an error when
+  // the kernel is launched.
+  const size_t MaxWorkGroupSize =
+      get_info<info::kernel_device_specific::work_group_size>(Device);
+  const size_t MaxLocalMemorySizeInBytes =
+      Device.get_info<info::device::local_mem_size>();
+
+  if (WorkGroupSize.size() > MaxWorkGroupSize)
+    return true;
+
+  if (DynamicLocalMemorySize > MaxLocalMemorySizeInBytes)
```

Comment:
Yeah absolutely. I am aware of this which is why I've already previous instances of device limits being used for max work-group size to kernel-device-specific. For this particular case, I am already using `kernel_device_specific::work_group_size` for the value of `MaxWorkGroupSize`, see:
```cpp
  const size_t MaxWorkGroupSize =
      get_info<info::kernel_device_specific::work_group_size>(Device);
```
from https://github.com/intel/llvm/pull/14333/files#diff-03b7a68ca0b70b58b49a0934a0d01b0f72ea187c3f8741e06aac5052ea20e961R115

Of course, do let me know if I am missing something. Thanks!

Reply:
Oh, right, it is already specific for a kernel. I guess I missed that because we are using `[implicit this.]get_info` instead of usual `kernel.get_info`